modules/sdk-api/src/v1/transactionBuilder.ts

```diff
@@ -802,16 +812,21 @@ const estimateTransactionSize = function (params) {
     throw new Error('expecting positive nOutputs');
   }
 
-  const estimatedSize =
-    VirtualSizes.txP2shInputSize * params.nP2shInputs +
+  // The size of an uncompressed public key is 32 bytes more than the compressed key,
+  // and hence, needs to be accounted for in the transaction size estimation.
+  const uncompressedPublicKeysTripleCorrectionFactor = 32 * 3;
+
+  return (
+    (VirtualSizes.txP2shInputSize +
+      (params.containsUncompressedPublicKeys ? uncompressedPublicKeysTripleCorrectionFactor : 0)) *
+      params.nP2shInputs +
```

Comment:
```suggestion
   // This is not quite accurate - if there is a mix of inputs scripts where some used
   // compressed keys and some used uncompressed keys, we would overestimate the size.
   // Since we don't have mixed input sets, this should not be an issue in practice.
    (VirtualSizes.txP2shInputSize +
      (params.containsUncompressedPublicKeys ? uncompressedPublicKeysTripleCorrectionFactor : 0)) *
      params.nP2shInputs +
```